src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Windows.cs

```diff
@@ -78,8 +78,30 @@ public FileSystemInfo ToFileSystemInfo()
         /// <summary>
         /// Returns the full path for find results, based on the initially provided path.
         /// </summary>
-        public string ToSpecifiedFullPath() =>
-            Path.Join(OriginalRootDirectory, Directory.Slice(RootDirectory.Length), FileName);
+        public string ToSpecifiedFullPath()
+        {
+            // We want to provide the enumerated segment of the path appended to the originally specified path. This is
+            // the behavior of the various Directory APIs that return a list of strings.
+            //
+            // RootDirectory has the final separator trimmed, OriginalRootDirectory does not. Our legacy behavior would
+            // effectively account for this by appending subdirectory names as it recursed. As such we need to trim one
+            // separator when combining with the relative path (Directory.Slice(RootDirectory.Length)).
+            //
+            //   Original  =>  Root   => Directory    => FileName => relativePath => Specified
+            //   C:\foo        C:\foo    C:\foo          bar         ""              C:\foo\bar
+            //   C:\foo\       C:\foo    C:\foo          bar         ""              C:\foo\bar
+            //   C:\foo/       C:\foo    C:\foo          bar         ""              C:\foo/bar
+            //   C:\foo\\      C:\foo    C:\foo          bar         ""              C:\foo\\bar
+            //   C:\foo        C:\foo    C:\foo\bar      jar         "bar"           C:\foo\bar\jar
+            //   C:\foo\       C:\foo    C:\foo\bar      jar         "bar"           C:\foo\bar\jar
+            //   C:\foo/       C:\foo    C:\foo\bar      jar         "bar"           C:\foo/bar\jar
+
+            ReadOnlySpan<char> relativePath = Directory.Slice(RootDirectory.Length);
+            if (PathInternal.EndsInDirectorySeparator(OriginalRootDirectory) && PathInternal.StartsWithDirectorySeparator(relativePath))
+                relativePath = relativePath.Slice(1);
+
+            return Path.Join(OriginalRootDirectory, relativePath, FileName);
+        }
```

Comment:
`ToSpecifiedFullPath` looks identical in both the Windows and Unix implementations.  Can it just be moved to a shared FileSystemEntry.cs location?

Reply:
Hmm- when I first did this I couldn't make a partial ref struct. I'll give it a shot.

Reply:
Ah, it is fields that I remembered. Can't spread them between files.